pkg/models/devops/devops.go

```diff
@@ -1009,3 +1014,118 @@ func parseBody(body io.Reader) (newReqBody io.ReadCloser) {
 	}
 	return rc
 }
+
+func (d devopsOperator) BuildPipelineParameters(projectName string, pipelineName string, query url.Values) ([]devopsv1alpha3.ParameterDefinition, error) {
+	projectObj, err := d.ksclient.DevopsV1alpha3().DevOpsProjects().Get(d.context, projectName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	piplineObj, err := d.ksclient.DevopsV1alpha3().Pipelines(projectObj.Status.AdminNamespace).Get(d.context, pipelineName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if piplineObj.Spec.Pipeline == nil {
+		return nil, fmt.Errorf("pipeline %s is not exist", pipelineName)
+	}
+	var allParams []devopsv1alpha3.ParameterDefinition
+	allParams = append(allParams, piplineObj.Spec.Pipeline.Parameters...)
+	for _, param := range piplineObj.Spec.Pipeline.ParametersFrom {
+		if param.Kind == "ConfigMap" {
+			if param.Mode == devopsv1alpha3.PARAM_REF_MODE_CONFIG {
+				params, err := buildParamFromConfigMapData(d.k8sclient, param.Name, param.ValuesKey)
+				if err != nil {
+					klog.Errorf("buildParamFromConfigMapData error:[%v]", err)
+					continue
+				}
+				allParams = append(allParams, params...)
+			} else if param.Mode == devopsv1alpha3.PARAM_REF_MODE_RESTFUL {
+				params, err := buildParamFromRESTfull(d.k8sclient, param.Name, query)
+				if err != nil {
+					klog.Errorf("buildParamFromRESTfull error:[%v]", err)
+					continue
+				}
+				allParams = append(allParams, params...)
```

Comment:
line 1040 and line 1047 are duplicated. Please feel free to consider put it into one line.